test/TestUtilities/Test.Utility/ProjectManagement/TestNuGetProjectContext.cs

```diff
@@ -20,14 +20,19 @@ public class TestNuGetProjectContext : IMSBuildNuGetProjectContext
         public Lazy<List<string>> Logs { get; } = new Lazy<List<string>>();
         public bool EnableLogging { get; set; }
 
+        private object _lock = new object();
+
         public void Log(MessageLevel level, string message, params object[] args)
         {
             // Uncomment when you want to debug tests.
             // Console.WriteLine(message, args);
 
             if (EnableLogging)
             {
-                Logs.Value.Add(args != null ? message + " " + string.Join(",", args) : message);
+                lock (_lock)
+                {
+                    Logs.Value.Add(args != null ? message + " " + string.Join(",", args) : message);
+                }
```

Comment:
I seriously doubt that this is a perf critical, especially since it's in test code, but still something worth considering for the future:

```suggestion
                string logMessage = args != null ? message + " " + string.Join(",", args) : message;
                lock (_lock)
                {
                    Logs.Value.Add(logMessage);
                }
```

This way the time required to do string concatenation does not extend the time that the lock is held.